contribute/design/1.x/cstor-operator/doc.md

```diff
@@ -544,19 +542,164 @@ spec:
+      overProvisioning: false
+      compression: false
+```
+
+Replacing block devices in mirror pool -- the spec will look
+following:
+
+```yaml
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-mirror
+spec:
+  pools:
+  - nodeSelector:
+      kubernetes.io/hostname: kubernetes-node1
+    raidGroups:
+    - type: mirror
+      isWriteCache: false
+      isSpare: false
+      isReadCache: false
+      blockDevices:
+
+      - blockDeviceName: node1-bd-3
+
+      - blockDeviceName: node1-bd-2
+
+    poolConfig:
+      cacheFile: /tmp/pool1.cache
       defaultRaidGroupType: mirror
       overProvisioning: false
       compression: false
 ```
+In the above CSPC spec node1-bd-1 is replaced with node1-bd-3
+
+Disk Replacement Validations:
+
+Once user modifies the CSPC spec to trigger disk replacement process below are
+the validations done by the admission-server to restrict invalid changes made to
+CSPC
+
+Below are validations on CSPC by admission server
+
+1. Not more than one block device should be replaced simultaneously in the same
+   raid group.
+
+2. Replacing block devices should not be already in use by the same cStor pools.
+
+3. Replacing another block device in the raid group is not allowed when any
+   of the block device in the same raid group is undergoing replacement.
+
+4. Steps to validate whether someone(other CSPC or local PV) already claimed
+   replacing block device
+
+   4.1 Verify is there any claim create for replacing block device. If there are
+       no claims for replacing block device jump to step 5.
+
+   4.2 If replacing block device was already claimed and if that claim doesn't
+       have CSPC label (or) different CSPC label(i.e block device belongs to
+       some other CSPC) then reject replacement request.
+
+   4.3 If replacing block device has self CSPC label then check is that replacing
+       block device is already linked with some other block device in that same
+       CSPC(i.e a case where users try to trigger a replace with disk that is
+       undergoing resilvering at cStor pool side). If link exists reject the request
+       else jump to step 5.
+
+5. Create a claim for replacing block device with annotation
+   `openebs.io/bd-predecessor: old_block_device_name` only if claim doesn't
+   exists.
+
+6. If claim already exists for replacing block device then update annotation
+   with proper block device name(Might be a case after creating claim update to
+   etcd got failed).
+
+`NOTE:`
+1. Pool expansion and disk replacement can go in parallel.
+
+2. Across the pool parallel block device replacements are allowed only when the
+   block devices belong to the different raid groups.
+
+Block Device Replacement Workflow: CSPC-Operator
+
+Work done by CStor-Operator after the user has updated the existing block device
+name with new block device name
+
+1. The CSPC-Operator will detect under which CSPC pool spec block device has been
```

Comment:
worth to mention that there can't be more than one change between CSPI and CPSC in a particular raid group

Reply:
Done